be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/DbActor.scala

```diff
@@ -168,84 +180,111 @@ object DbActor extends AskPatternConstants {
       channelDb
     }
 
-    override def receive: Receive = LoggingReceive {
-      case Write(channel, message) =>
-        log.info(s"Actor $self (db) received a WRITE request on channel '$channel'")
+    private def write(channel: Channel, message: Message): DbActorMessage = {
+      val channelDb: DB = channelsMap.get(channel) match {
+        case Some(channelDb) => channelDb
+        case _ => createDatabase(channel)
+      }
 
-        val channelDb: DB = channelsMap.get(channel) match {
-          case Some(channelDb) => channelDb
-          case _ => createDatabase(channel)
-        }
+      val messageId: Hash = message.message_id
+      Try(channelDb.put(messageId.getBytes, message.toJsonString.getBytes)) match {
+        case Success(_) =>
+          log.debug(s"Actor $self (db) wrote message_id '$messageId' on channel '$channel'")
+          DbActorWriteAck
+        case Failure(exception) =>
+          log.debug(s"Actor $self (db) encountered a problem while writing message_id '$messageId' on channel '$channel'")
```

Comment:
This looks like it might warrant a higher log level than `debug` , and we might want to print out the `exception` too